subgraph/src/listeners.ts

```diff
@@ -0,0 +1,80 @@
+import { Bytes } from '@graphprotocol/graph-ts'
+import {ERC20Mock as ERC20Contract} from '../types/ERC20/ERC20Mock'
+import { OfferCreated as OfferCreatedEvent } from '../types/OTC/OTC'
+import { Offer as OfferContract, OfferAccepted as OfferAcceptedEvent } from '../types/Offer/Offer'
+import { Offer } from '../types/schema'
+import { generateEvent, getOrCreateAsset, getAssetByAddress } from './utils'
+
+
+const getPrice = (address: string): i32 => {
+  const asset = getAssetByAddress(address)
+  const rate = asset.rate
+
+  if (!rate.startsWith('0x')) {
+    return i32(parseInt(rate))
+  }
+
+  return 0
+}
+
+export function handleOfferCreated(event: OfferCreatedEvent): void {
```

Comment:
I think we should handle events for adding more deposit into the same offer as well. This can be done later, after we launch it

Reply:
Sounds good